File: bukkit/src/main/java/io/github/eddiediamondfire/economyplus/command/SubCommand.java
package io.github.eddiediamondfire.economyplus.command;

import org.bukkit.entity.Player;

import java.util.List;

public interface SubCommand {

    String getName();
    String getDescription();
    String getSyntax();

    boolean action(Player player, String[] args);

    List<String> onTabComplete(Player player, String[] args);
}

File: server/src/main/kotlin/io/github/eddiediamondfire/economyplus/commands/SubCommand.kt
package io.github.eddiediamondfire.economyplus.commands

import org.bukkit.entity.Player
import org.jetbrains.annotations.NotNull

interface SubCommand {

    val name: String

    val syntax: String

    val description: String

    fun onCommand(player: Player, args: Array<String>): Boolean
}